_ingest-pipelines/processors/user-agent.md

```diff
@@ -0,0 +1,195 @@
+`ignore_failure` | Optional | Specifies whether the processor continues execution even if it encounters an error. If set to `true`, failures are ignored. Default is `false`. |
+`on_failure` | Optional | A list of processors to run if the processor fails. |
+`tag` | Optional | An identifier tag for the processor. Useful for debugging in order to distinguish between processors of the same type. |
+
+## Using the processor
+
+Follow these steps to use the processor in a pipeline.
+
+### Step 1: Create a pipeline
+
+The following query creates a pipeline named `user_agent_pipeline` that uses the `user_agent` processor to  to extract user agent information: 
+
+```json
+PUT _ingest/pipeline/user_agent_pipeline
+{
+  "description": "User agent pipeline",
+  "processors": [
+    {
+      "user_agent": {
+        "field": "user_agent",
+        "target_field": "user_agent_info"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+### Step 2 (Optional): Test the pipeline
+
+It is recommended that you test your pipeline before you ingest documents.
+{: .tip}
+
+To test the pipeline, run the following query:
+
+```json
+POST _ingest/pipeline/user_agent_pipeline/_simulate
+{
+  "pipeline": "user_agent_pipeline",
+  "docs": [
+    {
+      "_source": {
+        "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3"
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following example response confirms that the pipeline is working as expected:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "_index",
+        "_id": "_id",
+        "_source": {
+          "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3",
+          "user_agent_info": {
+            "name": "Chrome",
+            "original": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3",
+            "os": {
+              "name": "Windows",
+              "version": "10",
+              "full": "Windows 10"
+            },
+            "device": {
+              "name": "Other"
+            },
+            "version": "58.0.3029.110"
+          }
+        },
+        "_ingest": {
+          "timestamp": "2024-04-25T21:41:28.744407425Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+### Step 3: Ingest a document 
+
+The following query ingests a document into an index named `testindex1`:
+
+```json
+PUT testindex1/_doc/1?pipeline=user_agent_pipeline
+{
+  "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36"
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The request indexes the document into the index `testindex1` and will index all documents with the user agent string parsed into it components:
```

Comment:
This sentence needs a bit of revision. Please tag me on the rewrite.

Reply:
@natebower "The preceding request parses the `user_agent` string into its components and indexes the document, along with all documents containing those components, into the `testindex1` index."

Reply:
@natebower Thank you! Ready for your review and approval.